client/gutenberg/extensions/related-posts/edit.jsx

```diff
@@ -0,0 +1,123 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import classNames from 'classnames';
+import { __ } from '@wordpress/i18n';
+import { Fragment } from '@wordpress/element';
+import { BlockControls, InspectorControls } from '@wordpress/editor';
+import {
+	Button,
+	PanelBody,
+	RangeControl,
+	TextControl,
+	ToggleControl,
+	Toolbar,
+} from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import { DEFAULT_POSTS, MAX_POSTS_TO_SHOW } from './constants';
+
+export default ( { attributes, className, setAttributes } ) => {
+	const {
+		displayContext,
+		displayDate,
+		displayThumbnails,
+		headline,
+		postLayout,
+		postsToShow,
+	} = attributes;
+
+	const layoutControls = [
+		{
+			icon: 'grid-view',
+			title: __( 'Grid View' ),
+			onClick: () => setAttributes( { postLayout: 'grid' } ),
+			isActive: postLayout === 'grid',
+		},
+		{
+			icon: 'list-view',
+			title: __( 'List View' ),
+			onClick: () => setAttributes( { postLayout: 'list' } ),
+			isActive: postLayout === 'list',
+		},
+	];
+
+	const displayPosts =
+		DEFAULT_POSTS.length > postsToShow ? DEFAULT_POSTS.slice( 0, postsToShow ) : DEFAULT_POSTS;
+
+	return (
+		<Fragment>
+			<InspectorControls>
+				<PanelBody title={ __( 'Related Posts Settings' ) }>
+					<TextControl
+						label={ __( 'Headline' ) }
+						value={ headline }
+						onChange={ value => setAttributes( { headline: value } ) }
+					/>
+					<ToggleControl
+						label={ __( 'Display thumbnails' ) }
+						checked={ displayThumbnails }
+						onChange={ value => setAttributes( { displayThumbnails: value } ) }
+					/>
+					<ToggleControl
+						label={ __( 'Display date' ) }
+						checked={ displayDate }
+						onChange={ value => setAttributes( { displayDate: value } ) }
+					/>
+					<ToggleControl
+						label={ __( 'Display context (category or tag)' ) }
+						checked={ displayContext }
+						onChange={ value => setAttributes( { displayContext: value } ) }
+					/>
+					<RangeControl
+						label={ __( 'Number of posts' ) }
+						value={ postsToShow }
+						onChange={ value =>
+							setAttributes( { postsToShow: Math.min( value, MAX_POSTS_TO_SHOW ) } )
```

Comment:
nice `Math.min` here - it's easy to forget that we could get an update with a higher value if someone types it in!

Reply:
Shouldn't this kind of guard be upstreamed? I.e. `RangeControl` to not trigger `onChange` if `value` is greater than the `max` prop?

Reply:
I'll file a PR.

Reply:
Hmm, doesn't really work for `min` -- say your `min` is 10 and the user starts typing `15` -- the first keystroke (`1`) will trigger an `onChange` that will determine that `1<15` and thus prevent you from entering that value.

Reply:
> Hmm, doesn't really work for `min`

It works for our use case, where we're limited to enter values between 1 and 3.

This is also the reason why I didn't go for suggesting this to go upstream.